src/Planners/include/BipedalLocomotion/Planners/SO3Planner.h

```diff
@@ -0,0 +1,96 @@
+/**
+ * @file SO3Planner.h
+ * @authors Giulio Romualdi
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_PLANNERS_SO3_PLANNER_H
+#define BIPEDAL_LOCOMOTION_PLANNERS_SO3_PLANNER_H
+
+#include <manif/SO3.h>
+
+namespace BipedalLocomotion
+{
+namespace Planners
+{
+
+/**
+ * Representation used in the SO3Planner
+ */
+enum class Representation
+{
+    LeftTrivialized,
+    RightTrivialized
+};
+
+/**
+ * SO3Planner implements a minimum jerk trajectory planner for object belonging to SO(3). The
+ * planner assumes initial and final angular acceleration and velocity of the object equal to zero.
+ * The generated velocity and acceleration are written in the inertial or in body-fixed frame
+ * accordingly to the chosen representation.
+ * @tparam chosen representation:
+ * - The right trivialization planner generates a velocity and an acceleration expressed in the
+ * inertial frame.
+ * - The left trivialization planner generates a velocity and an acceleration expressed in the
+ * body-fixed frame.
+ */
+template <Representation representation> class SO3Planner
```

Comment:
Maybe, also this can inherit from ``Advanceable``?

Reply:
Working on this

Reply:
Done in 0942e9546c0e2b6633ecf3cd54c736d379810ca1. 

The thing I don't like is having a struct called `SO3PlannerState` (this is in general required by every object that inherits from `Advanceable`). Unfortunately, I don't think it is possible to do something like
```cpp
class SO3Planner : public Advanceable<???>
{
public:
   struct State
   {
         .... 
   };
};
```
probably using some forward declaration is possible. @S-Dafarra what do you thnk?

Reply:
If forward declaration works, I like that too, otherwise I think it is also not a big deal how it is now.

Reply:
This is a possible solution that uses the concept of `traits`. I already used it in [`lie-group-controllers`](https://github.com/dic-iit/lie-group-controllers). The effort required to change the code is not so high. I can start with that (in separate PR).
Let me know what do you think
```cpp
#include <iostream>

template<class T>
struct traits
{};

template<class T>
struct Advanceable
{
    using State = typename traits<T>::State;
    
    virtual void foo() const = 0;
};


class Planner;

template<>
struct traits<Planner>
{
    struct State 
    {
        double x{10};
    };
};

class Planner : public Advanceable<Planner>
{
public:
    using State = Advanceable::State;

private:
    State s;
    
public:
    void foo() const
    {
       std::cerr << s.x << std::endl;
    };
};


int main()
{
    Planner p;
    p.foo();
    Planner::State s;
    return 0;
}
```

Reply:
To be honest, I do not see any advantage of using this solution. In the end, you still have to declare the state outside the class. Am I missing something?

Reply:
Let me propose an alternative. What if we define a state like
```c++
template<class positionType, class velocityType, class accelerationType>
struct MechanicalState
{
    positionType position;
    velocityType velocity;
    accelerationType acceleration;
}
```
This one could go directly in ``Advanceable.h``, or in a header ``CommonStates.h`` in ``System``. Then, in this case, you would have 
```c++
class SO3Planner : public Advanceable<MechanicalState<manif::SO3d, manif::SO3d::Tangent, manif::SO3d::Tangent>>
{
    using State = MechanicalState<manif::SO3d, manif::SO3d::Tangent, manif::SO3d::Tangent>;
    ....
}
```

In general, we may reuse the state definitions somehow, they are not unique to the classes that use them.

Reply:
But yes I agree that probably now it's overkill!

Reply:
What do you think about https://github.com/dic-iit/bipedal-locomotion-framework/pull/84#discussion_r483015306?

Anyways, I agree that setting a state for each child of ``Advanceable`` may not be ideal, increasing also the probability of name clashes.

Reply:
> What do you think about #84 (comment)?

I completely missed the message, I'm sorry. 

Having `MechanicalState` it's a not a bad idea. The only limitation is the fact that it can be used only for a limited amount of objects. (e.g. SO3 planner, the QuinticSpline and the SwingFootPlanner). For instance, the `DCMPlanner` defines the state as: 
```cpp
struct State
{
    Eigen::Vector3d dcmPosition; /**< Position of the DCM expressed w.r.t. the inertial frame */
    Eigen::Vector3d dcmVelocity; /**< Velocity of the DCM expressed w.r.t. the inertial frame */
    Eigen::Vector3d vrpPosition; /**< Position of the virtual repellent point (VRP) expressed
                                      w.r.t. the inertial frame */
    double omega; /**< Value of the parameter omega */
 };
```
and also for the controllers, the `MechanicalState` may not be a suitable choice.

Anyways what you proposed could be a possible solution. Then if in the future we notice that are limitations we can still change the code. I can open a new PR as soon as this one, #83 and #90 are merged.  What do you think?

Reply:
> For instance, the `DCMPlanner` defines the state as:

Yes, I noticed it. But I think that it is fine if some classes have their own particular state definition. But if it is already ok for three classes, that would be already a plus.

> Then if in the future we notice that are limitations we can still change the code. I can open a new PR as soon as this one, #83 and #90 are merged. What do you think?

That would be great!